core/DataAccess/LogAggregator.php

```diff
@@ -1088,63 +1139,89 @@ public function queryConversionsByDimension($dimensions = array(), $where = fals
+
+        $dbSettings = new \Piwik\Db\Settings();
+        $tablePrefix = $dbSettings->getTablePrefix();
+
+        $sql = sprintf(
+        "SELECT
+          yyy.idvisit AS idvisit,
+          ".$idGoal." AS idgoal,
+          ".($linkField == 'idaction_url' ? '1' : '4')." AS `type`,
+          yyy.idaction AS idaction,
+          COUNT(*) AS `1`,     
+          " . sprintf("ROUND(SUM(yyy.revenue_total),2) AS `%d`,", Metrics::INDEX_GOAL_REVENUE) . "          
+          " . sprintf("COUNT(yyy.idvisit) AS `%d`,", Metrics::INDEX_GOAL_NB_VISITS_CONVERTED) . "                   
+          " . sprintf("ROUND(SUM(yyy.revenue_subtotal),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_SUBTOTAL) . "                 
+          " . sprintf("ROUND(SUM(yyy.revenue_tax),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_TAX) . "
+          " . sprintf("ROUND(SUM(yyy.revenue_shipping),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_SHIPPING) . "
+          " . sprintf("ROUND(SUM(yyy.revenue_discount),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_DISCOUNT) . "
+          " . sprintf("SUM(yyy.items) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_ITEMS) . "
+          " . sprintf("yyy.pages_before AS `%d`,", Metrics::INDEX_GOAL_NB_PAGES_UNIQ_BEFORE) . "                    
+          " . sprintf("SUM(yyy.attribution) AS `%d`,", Metrics::INDEX_GOAL_NB_CONVERSIONS_ATTRIB) . "                    
+          " . sprintf("COUNT(*) AS `%d`,", Metrics::INDEX_GOAL_NB_CONVERSIONS_PAGE_UNIQ) . "              
+          " . sprintf("ROUND(SUM(yyy.revenue),2) AS `%d`", Metrics::INDEX_GOAL_REVENUE_ATTRIB) . "         
+        FROM (
+          SELECT
+            num_total AS pages_before,
+            1 / num_total AS attribution,
+            r.idvisit AS idvisit,
+            lac.idaction AS idaction,
+            lvcon.revenue AS revenue_total,
+            1 / num_total * lvcon.revenue AS revenue,
+            1 / num_total * lvcon.revenue_subtotal AS revenue_subtotal,
+            1 / num_total * lvcon.revenue_tax AS revenue_tax,
+            1 / num_total * lvcon.revenue_shipping AS revenue_shipping,
+            1 / num_total * lvcon.revenue_discount AS revenue_discount,
+            1 / num_total * lvcon.items AS items
+          FROM (
+            %s
+          ) AS r
+          LEFT JOIN %slog_conversion lvcon ON lvcon.idgoal = ".$idGoal." AND lvcon.idvisit = r.idvisit
+          RIGHT JOIN %slog_link_visit_action logv ON logv.idvisit = r.idvisit
+          LEFT JOIN %slog_action lac ON logv.".$linkField." = lac.idaction
+          WHERE logv.server_time >= '%s'
+            AND logv.server_time <= '%s'
+            AND logv.idsite IN (%d) 
+            AND lac.type = ".($linkField == 'idaction_url' ? '1' : '4')."
+            AND logv.server_time <= lvcon.server_time
+          ) AS yyy
+        GROUP BY yyy.idaction
+        ORDER BY `9` DESC", $query['sql'], $tablePrefix, $tablePrefix, $tablePrefix,
```

Comment:
fyi for tablePrefix we usually use `Common::prefixTable('...table name...')` rather than concatenating it ourselves.